Introduce Target User

marcomc opened the pull request:
- Introduce the option to specify `target_user_id` to specify which user you want the applications to belong to, defaults to `ansible_user_id`
-- `target_user_id` feature required a better tasks grouping for different installer types
- replaces true/false with yes/no: yes/no are native to Ansible and do not require interpretation (fewer CPU cycles)

juju4 commented:
if this is about target user, please keep as target user only and move the rest separately.
something got travis broken
https://travis-ci.org/github/juju4/ansible-macos-apps-install/jobs/720443937#L14033

on true/false, please leave as is for now. ansible and yamllint are conflicting on that and for now, I'm more on the yamllint side.

Thanks

marcomc commented:
I have already fixed Travis, the issue was a wrong condition on a debug block.
I will revert the yes/no with True/False

marcomc commented:
> if this is about target user, please keep as target user only and move the rest separately.
reorganising the blocks was necessary for a clean implementation of the `target_user_id` feature

I have already fixed the TravisCI build
I have already reverted the yes/no to True/False (with capital letters is preferred to true/false because does not require interpretation, Ansible native is yes/no while True/False is YAML native)

there is nothing else going on in this PR

marcomc commented:
Hello @juju4 are you ok with all the corrections I made after your comments?

marcomc commented:
Hello again @juju4,
I really do not want to bother and pester you, so forgive me to insist: 
is there any chance you could review (and possible accept) this PR?

thanks for your understanding and patience

juju4 commented on tasks/archive-install.yml:
```diff
@@ -20,23 +20,19 @@
+  # we want the super-user to remvoe the existing app in case it was installed by another user
   - name: Remove existing app if present
     file:
       path: "{{ dest }}/{{app_basename}}"
```
please ensure spaces inside curly brackets (lint)

marcomc replied:
this is fixed, I also search for any other missing space after curly brackets

juju4 commented:
I'm usually less available during week days.

Comments
- need to add a test suite in travis. leave default as is and add a default-user with an alternate user created and role run with it. (matrix env suite+osx_image)
- still some true/false case change
- spaces in curly brackets variable per ansible-lint in a few places

Thanks

marcomc commented:
Thank you, Julien
I did all you asked, now I am waiting to see if build #320 will complete
successfully.
nonetheless, I see that build #319 says it is started since 31 and more
minutes, but you can s see that none of jobs have actually started
I do not understand if this is an issue with the GUI of Travis or some
error in my travis settings

if you have time give it a look and tell me what you think.

thank you

Marco

On Sat, 29 Aug 2020 at 10:20, Julien <notifications@github.com> wrote:

> I'm usually less available during week days.
>
> Comments
>
>    - need to add a test suite in travis. leave default as is and add a
>    default-user with an alternate user created and role run with it. (matrix
>    env suite+osx_image)
>    - still some true/false case change
>    - spaces in curly brackets variable per ansible-lint in a few places
>
> Thanks
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/juju4/ansible-macos-apps-install/pull/2#issuecomment-683289240>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAK4ZSBGIAXJJ53V747TTV3SDD6BFANCNFSM4QI2AXUA>
> .
>


-- 
Marco Massari Calderone

juju4 commented:
Please try a different archives folder like ~/tmp
Catalina constraint if using known folders... https://nektony.com/duplicate-finder-free/folders-permission

https://travis-ci.org/github/juju4/ansible-macos-apps-install/builds/722353093#L9588
```
OSError: [Errno 1] Operation not permitted: '/Users/travis/Downloads'
```

for other release, it seems hdiutil related because trying to mount from wrong place: download to /var/root/Downloads and trying to mount $HOME/Downloads/. likely $HOME not expanded.
https://travis-ci.org/github/juju4/ansible-macos-apps-install/jobs/722337101#L15877

as for the extra user, for the limited scope of test, I would just add a user: module call in the test playbook instead of adding an extra role call.

marcomc commented:
Hello Julien

all the 6 Travis builds are completing successfully now.
I have implemented a task to create a new test user and remove the extra
role.

please review the PR when you can and let me know if think more adjustments
need to be made

thanks for your help

Marco

On Sat, 29 Aug 2020 at 18:27, Julien <notifications@github.com> wrote:

> Please try a different archives folder like ~/tmp
> Catalina constraint if using known folders...
> https://nektony.com/duplicate-finder-free/folders-permission
>
>
> https://travis-ci.org/github/juju4/ansible-macos-apps-install/builds/722353093#L9588
>
> OSError: [Errno 1] Operation not permitted: '/Users/travis/Downloads'
>
> for other release, it seems hdiutil related because trying to mount from
> wrong place: download to /var/root/Downloads and trying to mount
> $HOME/Downloads/. likely $HOME not expanded.
>
> https://travis-ci.org/github/juju4/ansible-macos-apps-install/jobs/722337101#L15877
>
> as for the extra user, for the limited scope of test, I would just add a
> user: module call in the test playbook instead of adding an extra role call.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/juju4/ansible-macos-apps-install/pull/2#issuecomment-683343847>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAK4ZSEMELEV6JTULMWZMBLSDFXETANCNFSM4QI2AXUA>
> .
>


-- 
Marco Massari Calderone

juju4 commented:
Great. Thanks for the work @marcomc!

Just that if I checked targetuser suite
https://travis-ci.org/github/juju4/ansible-macos-apps-install/jobs/722393344#L9991
```
TASK [juju4.macos_apps_install : debug] ****************************************
task path: /Users/travis/build/juju4/ansible-macos-apps-install/tasks/archive-install.yml:62
ok: [localhost] => {
    "msg": "This app will be installed as the user `targetuser`"
}
```

but installation is in /Applications, not ~/Applications
https://travis-ci.org/github/juju4/ansible-macos-apps-install/jobs/722393344#L11415
```
changed: [localhost] => {
    "changed": true,
    "dest": "/Applications/",
    "invocation": {
        "dest": "/Applications/",
        "module_args": {
            "dest": "/Applications/",
            "owner": "targetuser",
            "src": "/Volumes/Firefox/Firefox.app"
        },
        "owner": "targetuser",
        "src": "/Volumes/Firefox/Firefox.app"
    },
    "src": "/Volumes/Firefox/Firefox.app"
}
```

or was the intention was only for permissions and not destination path?
else all good

marcomc commented:
It's intended: I just wanted the ownership to be the one of the target user so the time the target user wants to update the application manually he won't be asked for authentication

juju4 commented:
ok. LGTM.
may want to add path options but for another time. Thanks!

marcomc commented:
Thanks for your support Julien

On Sun, 30 Aug 2020 at 13:15, Julien <notifications@github.com> wrote:

> Merged #2 <https://github.com/juju4/ansible-macos-apps-install/pull/2>
> into master.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/juju4/ansible-macos-apps-install/pull/2#event-3707430515>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAK4ZSCODIUXB4QGEOUNUODSDJ3LLANCNFSM4QI2AXUA>
> .
>


-- 
Marco Massari Calderone

marcomc commented:
Hello Julien, how are you doing?


In the past months, I have been working to create a tool called `Splinter`
(like the ninja rat) that I wanted to introduce to you because it makes
important use of your https://github.com/juju4/ansible-macos-apps-install and
possibly you might be interested in trying yourself this new tool of mine.


This is the GitHub home of Splinter: https://github.com/marcomc/splinter/


Splinter is a provisioning tool based on BASH and Ansible.

- It takes about 15-20 minutes to create a reusable configuration (a
profile).

- Then it tales ~1h 15m for fully unattended provisioning.


The outcome is a Mac with (everything is modular and optional)

- an `admin` account for the use of the support team

- a new account that will request the user to change its password at the
first login

- Filevault encrypted filesystem

- Sophos AV Endpoint installed

- All the desired applications preinstalled (i.e. Chrome, Atom, MS Office,
Virtualbox - you choose what)

- a user's favourite system config and user preferences and dotfiles

- a user's chose PIP, NPM, Ruby, Homebrew packages and formulas

- .ssh_config already set to use proxy gateways

... and much much more


Among the many things it can do:

- the automated installation of Sophos AV Endpoint installation

- the automated activation of FileVault2 encryption with personal or
institutional recovery keys

- the installation of Homebrew packages and Cask formulas

- setup of common settings to be set across the whole company or to keep
separate configurations for your family Macs

- setup of role-specific settings for each department, person or Mac


Splinter can be safely installed in your machine, it will not clutter your
system because everything that it uses lives inside the project directory
created during the installation (removing the project directory removes
everything that has been installed). In fact, all the Python requirements
for Ansible are prepacked in a Miniconda (or Pyenv) environment that lives
within the Splinter project directory.


Once installed in your machines, you can configure and create a
self-contained provisioning package to copy to the machine to provision.

Creating a provisioning package is not mandatory; you can run provisioning
directly on your target machine on the fly if you store your custom
profiles in a GitHub repository.


I have documented all its capabilities in here:
https://github.com/marcomc/splinter/wiki


You will probably find most helpful the step-by-step guide I wrote to
migrate to a new Mac:
https://github.com/marcomc/splinter/wiki/Migrate-to-a-new-Mac


I would really appreciate if you could give it a look and send me your
feedback and impressions.

Thank you for your excellent work on Ansible staff


Marco MC


On Sun, 30 Aug 2020 at 14:12, Marco Massari Calderone <massaric@gmail.com>
wrote:

> Thanks for your support Julien
>
> On Sun, 30 Aug 2020 at 13:15, Julien <notifications@github.com> wrote:
>
>> Merged #2 <https://github.com/juju4/ansible-macos-apps-install/pull/2>
>> into master.
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/juju4/ansible-macos-apps-install/pull/2#event-3707430515>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AAK4ZSCODIUXB4QGEOUNUODSDJ3LLANCNFSM4QI2AXUA>
>> .
>>
>
>
> --
> Marco Massari Calderone
>
>
>
>

-- 
Marco Massari Calderone
